### PR TITLE
Games api and card/feature/#3

### LINKED ISSUE #3: Make api calls and game card

### PR DESCRIPTION
**Pull request by drmodun**
closes #3 
should be done, netlify seems to be having some trouble doing ISR  and SSR functions, will try to fix it but I am pretty sure it is their bug (https://answers.netlify.com/t/next-js-app-directory-some-pages-are-500-internal-server-error-on-deployment/93203)

### CODE REVIEW

**Comment by drmodun** [comment]
good news and bad news. I have managed to fix the error by changing the next js version, but because it is downgraded to 13.2 some optimisations and good fetaures might be missing, currently this is the best option, currently deployed (https://gamesapiandcard-feature-3--pub-critic.netlify.app/)

**Comment by ilovric00** [comment]
> good news and bad news. I have managed to fix the error by changing the next js version, but because it is downgraded to 13.2 some optimisations and good fetaures might be missing, currently this is the best option, currently deployed (https://gamesapiandcard-feature-3--pub-critic.netlify.app/)

That version doesn't have stable app router and some of the features are experimental

File: pub-critic/src/app/api/GamesApi.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+};
+
+export const getLatestGames = async (
+  page: number = 8,
+  pageSize: number = 20
+) => {
+  try {
+    const response = await fetch(
+      `${gamesApi}/games?key=${apiKey}&dates=2023-01-01&page=${page}&page_size=${pageSize}&ordering=-released`,
```
Comment (ilovric00) [inline pub-critic/src/app/api/GamesApi.ts]: https://www.npmjs.com/package/query-string can simplify these query params you have

File: pub-critic/src/app/api/GamesShared.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  name: string;
+  released: string;
+  background_image: string;
+  metacritic: number;
+  rating: number;
+  short_screenshots: Screenshot[];
+  tags: Tag[];
+  genres: Genre[];
+  platforms: PlatformFull[];
+}
```
Comment (ilovric00) [inline pub-critic/src/app/api/GamesShared.ts]: create `app/interfaces` folder and move all common interfaces there

**Comment by drmodun** [comment]
> > good news and bad news. I have managed to fix the error by changing the next js version, but because it is downgraded to 13.2 some optimisations and good fetaures might be missing, currently this is the best option, currently deployed (https://gamesapiandcard-feature-3--pub-critic.netlify.app/)
> 
> That version doesn't have stable app router and some of the features are experimental

I ended up using 13.4 again and switched from netlify to vercel because netlify didnt support server side search params, I believe that the other branches should be accessible on vercel but if not I will post the links

File: pub-critic/src/app/components/GameCard/GameCard.module.scss [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    display: flex;
+
+    @media (max-width: 768px) {
+      width: vw-mobile-calc(160);
+      height: vw-mobile-calc(150);
+      border-radius: vw-mobile-calc(8) vw-mobile-calc(8) 0 0;
+    }
+    width: vw-calc(460);
+    height: vw-calc(450);
+    border-radius: vw-calc(8) vw-calc(8) 0 0;
```
Comment (ilovric00) [inline pub-critic/src/app/components/GameCard/GameCard.module.scss]: `vw-calc(8 8 0 0);` it's the same 🙂

File: pub-critic/src/app/components/GameCard/GameCard.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import { Game, Platform, PlatformFull } from "api/GamesShared";
+import classes from "./GameCard.module.scss";
+import Image from "next/image";
+import playstation from "assets/playstation.svg";
+import xbox from "assets/xbox.svg";
+import nintendo from "assets/nintendo.svg";
+import epic from "assets/epic.svg";
+import steam from "assets/steam.svg";
+import controller from "assets/controller.svg";
+interface GameCardProps {
```
Comment (ilovric00) [inline pub-critic/src/app/components/GameCard/GameCard.tsx:7]: space between

File: pub-critic/src/app/components/GameCard/GameCard.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  if (
+    platforms.find((p) => p.platform.name.toLowerCase().includes("nintendo"))
+  ) {
+    images.push(nintendo);
+  }
+  if (platforms.find((p) => p.platform.name.toLowerCase().includes("epic"))) {
+    images.push(epic);
+  }
+  return images;
+};
```
Comment (ilovric00) [inline pub-critic/src/app/components/GameCard/GameCard.tsx]: 1) tooooo many ifs, this must be simplified somehow in way that you find a platform name that matches certain platform -> `function` with "platform" `param`
2) move this to `app/utils/static`

File: pub-critic/src/app/page.module.scss [hunk 6]
```diff
@@ -102,6 +103,39 @@
...
+
+      .list {
+        width: 100%;
+        display: flex;
+        flex-direction: row;
+        align-items: flex-start;
+        justify-content: flex-start;
+        overflow-x: scroll;
+        margin-top: 20px;
+        > div {
```
Comment (ilovric00) [inline pub-critic/src/app/page.module.scss]: Would be better to use class selector than this way

**Review by ilovric00 — CHANGES_REQUESTED** [review]
Good work in general, just a few small changes and you are good to go 👏

File: pub-critic/src/app/api/GamesApi.ts (hunk 1 above)
Comment (drmodun) [reply pub-critic/src/app/api/GamesApi.ts]: :white_check_mark:

File: pub-critic/src/app/api/GamesShared.ts (hunk 2 above)
Comment (drmodun) [reply pub-critic/src/app/api/GamesShared.ts]: :white_check_mark:

File: pub-critic/src/app/components/GameCard/GameCard.module.scss (hunk 3 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameCard/GameCard.module.scss]: :white_check_mark:

File: pub-critic/src/app/components/GameCard/GameCard.tsx (hunk 4 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameCard/GameCard.tsx:7]: :white_check_mark:

File: pub-critic/src/app/page.module.scss (hunk 6 above)
Comment (drmodun) [reply pub-critic/src/app/page.module.scss]: :white_check_mark:

File: pub-critic/src/app/components/GameCard/GameCard.tsx (hunk 5 above)
Comment (drmodun) [reply pub-critic/src/app/components/GameCard/GameCard.tsx]: :white_check_mark:

File: pub-critic/src/app/components/sideMenu/SideMenu.tsx [hunk 7]
```diff
@@ -5,19 +5,24 @@ import classes from "./SideMenu.module.scss";
...
   active: boolean;
+  user?: boolean;
 }
 
-export const SideMenu = ({ toggleMenu, active }: SideMenuProps) => {
+export const SideMenu = ({ toggleMenu, active, user }: SideMenuProps) => {
   return (
-    <div className={classes.sideMenu}>
+    <div
+      className={`${classes.sideMenu} ${active ? classes.active : ""}`}
```
Comment (ilovric00) [inline pub-critic/src/app/components/sideMenu/SideMenu.tsx:14]: https://github.com/lukeed/clsx

File: pub-critic/src/app/page.module.scss [hunk 8]
```diff
@@ -1,4 +1,5 @@
 @import "styles/variables.scss";
+@import "styles/vw-calc.scss";
 .container {
```
Comment (ilovric00) [inline pub-critic/src/app/page.module.scss:3]: space after import statements

**Review by ilovric00 — APPROVED** [review]
just a minor improvements, if you want to learn more :)

File: pub-critic/src/app/components/sideMenu/SideMenu.tsx (hunk 7 above)
Comment (drmodun) [reply pub-critic/src/app/components/sideMenu/SideMenu.tsx:14]: yep gonna use clsx in the future for this, might change it later if I have time

File: pub-critic/src/app/page.module.scss (hunk 8 above)
Comment (drmodun) [reply pub-critic/src/app/page.module.scss:3]: ✅

**Pull request merged**